api/attachments_api.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2024 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License")
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from chromestatus_openapi.models import AddAttachmentResponse
+
+from framework import basehandlers
+from framework import permissions
+from internals import attachments
+
+
+class AttachmentsAPI(basehandlers.EntitiesAPIHandler):
+  """Features are the the main records that we track."""
+
+  @permissions.require_create_feature
+  def do_post(self, **kwargs) -> dict[str, str]:
+    """Handle POST requests to create a single feature."""
+    feature_id = kwargs.get('feature_id', None)
+
+    # Validate the user has edit permissions and redirect if needed.
+    redirect_resp = permissions.validate_feature_edit_permission(
+        self, feature_id)
+    if redirect_resp:
+      self.abort(403, msg='User lacks permission to edit')
+
+    files = kwargs.get('mock_files', self.request.files)
```

Comment:
We should not have this logic that looks for mock_files in the keywords in the live code. Instead, we should adjust the tests. More about this in the test file.
```suggestion
    files = self.request.files
```